lib/handler/status/requests.c

```diff
@@ -71,9 +72,11 @@ static void requests_status_per_thread(void *priv, h2o_context_t *ctx)
     if (cbdata.buffer->size != 0) {
         if (rsc->req_data.len == 0)
             h2o_buffer_consume(&cbdata.buffer, 1); /* skip preceeding comma */
+        pthread_mutex_lock(&rsc->mutex);
```

Comment:
I believe this needs to be moved above the `rsc->req_data.len` check, otherwise we might race and skip the coma multiple times.